Code/Source/Manipulator/MotorizedJoint.cpp

```diff
@@ -0,0 +1,209 @@
+                ->Field("TestSinAmplitude", &MotorizedJoint::m_sin_amplitude)
+                ->Field("TestSinFreq", &MotorizedJoint::m_sin_freq)
+                ->Field("DebugPrint", &MotorizedJoint::m_debug_print);
+
+            if (AZ::EditContext* ec = serialize->GetEditContext())
+            {
+                ec->Class<MotorizedJoint>("MotorizedJoint", "MotorizedJoint")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "MotorizedJoint")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC("Game"))
+                    ->Attribute(AZ::Edit::Attributes::Category, "MotorizedJoint")
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &MotorizedJoint::m_joint_dir, "Dir.", "Direction of joint.")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &MotorizedJoint::m_limits,
+                        "ControllerLimits",
+                        "When measurment is outside the limits, ")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &MotorizedJoint::m_debug_draw_entity,
+                        "Setpoint",
+                        "Allows to apply debug setpoint visualizer")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &MotorizedJoint::m_zero_offset,
+                        "Zero Off.",
+                        "Allows to change offset of zero to set point")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default, &MotorizedJoint::m_linear, "Apl. Linear", "Applies linear force instead of torque")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &MotorizedJoint::m_animation_mode,
+                        "Animation mode",
+                        "In animation mode, the transform API is used instead of Rigid Body. Note that using this "
+                        "the Rigid Body Component should be disabled.")
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &MotorizedJoint::m_pid_pos_conf, "PidPosition", "PidPosition")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &MotorizedJoint::m_test_sinusoidal,
+                        "SinusoidalTest",
+                        "Allows to apply sinusoidal test signal")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default, &MotorizedJoint::m_sin_amplitude, "Amplitude", "Amplitude of sinusoidal test signal")
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &MotorizedJoint::m_sin_freq, "Frequency", "TestSinFreq")
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &MotorizedJoint::m_debug_print, "Debug", "Print debug to console");
```

Comment:
I guess this is a temporary solution, which is ok

Reply:
Yes. I strongly recommend using IMGui here.